ledger/ledger-api-integration-tests/src/test/lib/scala/com/digitalasset/platform/apitesting/RemoteServerResource.scala

```diff
@@ -49,16 +49,21 @@ class RemoteServerResource(host: String, port: Int, tlsConfig: Option[TlsConfigu
 
   }
 
-  def createEventLoopGroup(threadPoolName: String): NioEventLoopGroup = {
+  private def createEventLoopGroup(threadPoolName: String): NioEventLoopGroup = {
     val threadFactory = new DefaultThreadFactory(s"$threadPoolName-grpc-eventloop", true)
     val parallelism = Runtime.getRuntime.availableProcessors
     new NioEventLoopGroup(parallelism, threadFactory)
   }
 
   override def close(): Unit = {
     channel.shutdownNow()
-    channel.awaitTermination(1L, TimeUnit.SECONDS)
-    eventLoopGroup.shutdownGracefully().await(1L, TimeUnit.SECONDS)
+    if (!channel.awaitTermination(10L, TimeUnit.SECONDS)) {
+      throw new RuntimeException(
```

Comment:
use `sys.error` instead.

Reply:
note, that with throwing exceptions you shortcut the program logic, and will miss closing further resources, possibly making an even bigger problem to what you already have.